File: single-language-analysis/dataset-files/python/test_smell_142.py
# Original URL: https://github.com/ogrisel/scikit-learn/blob/b20998d4ec606558ae86cba7ff5a84aea094c397/sklearn/cluster/tests/test_k_means.py#L770-L781

def test_smell_142(global_random_seed):
    # test calling the k_means function directly
    cluster_centers, labels, inertia = k_means(
        X, n_clusters=n_clusters, sample_weight=None, random_state=global_random_seed
    )

    assert cluster_centers.shape == (n_clusters, n_features)
    assert np.unique(labels).shape[0] == n_clusters

    # check that the labels assignment are perfect (up to a permutation)
    assert_allclose(v_measure_score(true_labels, labels), 1.0)
    assert inertia > 0.0

File: single-language-analysis/dataset-files/java/TestSmell142.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell142 {

    @Test
    public void testKMeansFunction() {
        // test calling the k_means function directly
        double[][] clusterCenters = new double[nClusters][nFeatures];
        int[] labels = new int[X.length];
        double inertia = kMeans(X, nClusters, null, globalRandomSeed, clusterCenters, labels);

        assertEquals(new int[]{nClusters, nFeatures}, clusterCenters.shape());
        assertEquals(nClusters, unique(labels).length);

        // check that the labels assignment are perfect (up to a permutation)
        assertEquals(1.0, vMeasureScore(trueLabels, labels), 0.0);
        assertTrue(inertia > 0.0);
    }
}